salt/cloud/clouds/linode.py

```diff
@@ -1,38 +1,118 @@
-also needs to be set. The password needs to be 8 characters and contain lowercase, uppercase, and numbers.
+- **apikey**: (required) The key to use to authenticate with the Linode API.
+- **password**: (required) The default password to set on new VMs. Must be 8 characters with at least one lowercase, uppercase, and numeric.
+- **api_version**: (optional) The version of the Linode API to interact with. Defaults to ``v3``.
+- **poll_interval**: (optional) The rate of time in milliseconds to poll the Linode API for changes. Defaults to ``500``.
+- **ratelimit_sleep**: (optional) The time in seconds to wait before retrying after a ratelimit has been enforced. Defaults to ``0``.
 
-Set up the cloud configuration at ``/etc/salt/cloud.providers`` or ``/etc/salt/cloud.providers.d/linode.conf``:
+.. note::
+
+    APIv3 usage is deprecated and will be removed in a future release in favor of APIv4. To move to APIv4 now,
+    set the ``api_version`` parameter in your provider configuration to ``v4``. See the full migration guide
+    here https://docs.saltstack.com/en/latest/topics/cloud/linode.html#migrating-to-apiv4.
+
+Set up the provider configuration at ``/etc/salt/cloud.providers`` or ``/etc/salt/cloud.providers.d/linode.conf``:
 
 .. code-block:: yaml
 
     my-linode-provider:
-      apikey: f4ZsmwtB1c7f85Jdu43RgXVDFlNjuJaeIYV8QMftTqKScEB2vSosFSr...
-      password: F00barbaz
-      driver: linode
+        driver: linode
+        api_version: v4
+        apikey: f4ZsmwtB1c7f85Jdu43RgXVDFlNjuJaeIYV8QMftTqKScEB2vSosFSr...
+        password: F00barbaz
 
-    linode-profile:
-      provider: my-linode-provider
-      size: Linode 1024
-      image: CentOS 7
-      location: London, England, UK
+For use with APIv3 (deprecated):
 
-"""
+.. code-block:: yaml
 
-# Import Python Libs
-from __future__ import absolute_import, print_function, unicode_literals
+    my-linode-provider-v3:
+        driver: linode
+        apikey: f4ZsmwtB1c7f85Jdu43RgXVDFlNjuJaeIYV8QMftTqKScEB2vSosFSr...
+        password: F00barbaz
+
+Profile
+-------
+
+The following profile parameters are supported:
+
+- **size**: (required) The size of the VM. This should be a Linode instance type ID (i.e. ``g6-standard-2``). For APIv3, this would be a plan ID (i.e. ``Linode 2GB``). Run ``salt-cloud -f avail_sizes my-linode-provider`` for options.
+- **location**: (required) The location of the VM. This should be a Linode region (e.g. ``us-east``). For APIv3, this would be a datacenter location (i.e. ``Newark, NJ, USA``). Run ``salt-cloud -f avail_locations my-linode-provider`` for options.
+- **image**: (required) The image to deploy the boot disk from. This should be an image ID (e.g. ``linode/ubuntu16.04``); official images start with ``linode/``. For APIv3, this would be an image label (i.e. Ubuntu 16.04). Run ``salt-cloud -f avail_images my-linode-provider`` for more options.
+- **password**: (\*required) The default password for the VM. Must be provided at the profile or provider level.
```

Comment:
what's with the `\*required` ?

Reply:
> Must be provided at the profile or provider level.
I was trying to convey that a `password` value is required, but not necessarily at the VM configuration level.

Reply:
I guess that works.